pkg/workloads/console/controller.go

```diff
@@ -457,12 +516,16 @@ func calculateStatus(csl *workloadsv1alpha1.Console, job *batchv1.Job, pod *core
 		newStatus.PodName = pod.ObjectMeta.Name
 	}
 
-	newStatus.Phase = calculatePhase(job, pod)
+	newStatus.Phase = calculatePhase(authorised, job, pod)
 
 	return newStatus
 }
 
-func calculatePhase(job *batchv1.Job, pod *corev1.Pod) workloadsv1alpha1.ConsolePhase {
+func calculatePhase(authorised bool, job *batchv1.Job, pod *corev1.Pod) workloadsv1alpha1.ConsolePhase {
+	if !authorised {
```

Comment:
I reckon this block should go down to just before where we return `ConsolePending`. Assuming no programming errors it's maybe irrelevant, but regardless I think that all of the other conditions that we check for should take precedence before this one?

Reply:
We aren't able to move this as the function is written but we would still like to improve this function. As it is currently written it assumes that the input arguments are correct, but if they aren't it can have odd behaviour and return the wrong phase.